src/TestingUtils/Microsoft.AspNetCore.Testing/src/xunit/AspNetTestAssemblyRunner.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    public class AspNetTestAssemblyRunner : XunitTestAssemblyRunner
```

Comment:
OK, you have reached **the fun part**.

There's like 9 levels of extensibility we need to implement between the top level (framework), and the lowest level (invoker). Most of them are just pass-through.